src/query/src/part_sort.rs

```diff
@@ -338,16 +375,42 @@ impl PartSortStream {
         Ok(None)
     }
 
+    fn push_buffer(&mut self, batch: DfRecordBatch) -> datafusion_common::Result<()> {
+        match &mut self.buffer {
+            PartSortBuffer::All(v) => v.push(batch),
+            PartSortBuffer::Top(top, cnt) => {
+                *cnt += batch.num_rows();
```

Comment:
Looks like the row count number records how many rows were pushed into this buffer, but maybe it is not precise because the `TopK` only keeps top `k` rows. Am I right?

Reply:
Yes, this `cnt` might be greater than `k`. It's usage is to check if the buffer is empty. I've updated the doc comment.